File: AD-HOC/2443/main.dart
import 'dart:io';

void printQuotient(int a, int b, int c, int d) {
  int numerator = a * d + b * c;
  int denominator = b * d;
  int gcd = numerator.gcd(denominator);
  print('${numerator ~/ gcd} ${denominator ~/ gcd}');
}

void main() {
  var [a, b, c, d] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  printQuotient(a, b, c, d);
}

File: AD-HOC/2443/main.js

function printQuotient(a, b, c, d) {
    let numerator = a * d + b * c;
    let denominator = b * d;
    let gcd = _gcd(numerator, denominator);
    console.log(`${numerator / gcd} ${denominator / gcd}`);
}

function _gcd(x, y) {
    return y === 0 ? x : _gcd(y, x % y);
}

let [a, b, c, d] = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split(' ').map(Number);
printQuotient(a, b, c, d);
